plugins/user-authenticators/ldap/src/org/apache/cloudstack/api/command/LdapCreateAccountCmd.java

```diff
@@ -119,6 +131,9 @@ private Long getDomainId() {
 
     @Override
     public void execute() throws ServerApiException {
+        if (getAccountType() == null && getRoleId() == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Both account type and role ID are not provided");
```

Comment:
@jburwell because of backward compatibility we cannot make roleid param as required, while accounttype used to be required but is now made non-required; this checks ensures that at least one of them is passed. When only account type is passed, we translate that to one of the four default roles (id).